include/Zydis/Internal/SharedData.h

```diff
@@ -716,6 +686,21 @@ typedef enum ZydisMaskOverride_
 
 /* ---------------------------------------------------------------------------------------------- */
 
+#define ZYDIS_OPDEF_REQUIRED_BITS \
+    ZYAN_MAX(ZYDIS_REGKIND_REQUIRED_BITS, ZYDIS_MEMOP_TYPE_REQUIRED_BITS + 1) + 1
+
+#define ZYDIS_OPDEF_GET_REG(operand_definition) \
+    ((operand_definition) & ((1 << ZYDIS_REGKIND_REQUIRED_BITS   ) - 1))
+
+#define ZYDIS_OPDEF_GET_MEM(operand_definition) \
+    ((operand_definition) & ((1 << ZYDIS_MEMOP_TYPE_REQUIRED_BITS) - 1))
+
+#define ZYDIS_OPDEF_GET_REG_HIGH_BIT(operand_definition) \
+    (((operand_definition) >> ZYDIS_REGKIND_REQUIRED_BITS   ) & 0x01)
+
+#define ZYDIS_OPDEF_MEM_GET_HIGH_BIT(operand_definition) \
```

Comment:
For naming consistency:
```suggestion
#define ZYDIS_OPDEF_GET_MEM_HIGH_BIT(operand_definition) \
```